### PR TITLE
Implementation of restrictions, validations and custom messages

### LINKED ISSUE #6: Template restrictions
**Restrictions to add**
Add restrictions to crud operations based on requirements, check trello for more information

### PR DESCRIPTION
**Pull request by coby111**
## Overview
This PR introduces the following improvements and functionalities to the template service:

**Template creation:**
 - Updated the method to create templates (`create`).
 - Validation of foreign keys.
 - Exception handling with custom messages.

**List of templates:**
 - Updated the method to list all templates (`findAll`).
 - Exception handling with custom messages.

 **Template update:**
 - Update of the method to update templates (`update`).
 - Period validation.
 - Exception handling with custom messages.

 **Deletion of templates:**
 - Updated the method to remove templates (`remove`).
 - Exception handling with custom messages.

**Validation of IDs and foreign keys:**
 - Added method to validate existing template ID (`validateId`).
 - Added method to validate foreign keys (`validateAreaId`, `validateResponsibleId`, `validateRevisedById`).

**DTOs with validations:**
 - Updated `CreateTemplateDto` with validations and custom messages.
 - Updated `UpdateTemplateDto` with validations and custom messages.

### Service

- **create**
- **findAll**
- **update**
- **remove**
- **validateId**
- **validateAreaId**
- **validateRevisedById**
- **validateResponsibleId**

### DTOs

- **CreateTemplateDto**
- **UpdateTemplateDto**

## How to test

1. **Template creation:**
 - Send a POST request to `/templates` with valid and invalid data to check validations and error messages.

2. **List of templates:**
 - Send a GET request to `/templates` to check listing and exception handling.

3. **Template update:**
 - Send a PATCH request to `/templates/:id` with valid and invalid data to check validations and error messages.

4. **Template removal:**
 - Send a DELETE request to `/templates/:id` to check for deletion and exception handling.

## Related Issues
- Closes #6 

Thank you for reviewing this pull request. I look forward to your feedback.

### CODE REVIEW

File: src/models/template/create-template.dto.ts [hunk 1]
```diff
@@ -1,7 +1,32 @@
+import { IsInt, IsNotEmpty, IsString, Min } from 'class-validator';
```
Comment (DiegoSHS) [inline src/models/template/create-template.dto.ts:1]: Change the messages for responsibleId and revisedId, avoid yo use the word "id"

File: src/models/template/update-template.dto.ts [hunk 2]
```diff
@@ -1,10 +1,37 @@
-import { PartialType } from '@nestjs/mapped-types';
-import { CreateTemplateDto } from './create-template.dto';
+import { IsInt, IsNotEmpty, IsOptional, IsString, Min } from 'class-validator';
```
Comment (DiegoSHS) [inline src/models/template/update-template.dto.ts:1]: Same for the UpdateDTO, avoid using the word "ID" for those fields

**Review by DiegoSHS — CHANGES_REQUESTED** [review]
Check the comments, just those changes and that's it

**Review by DiegoSHS — APPROVED** [review]
Reviewed again, good changes

**Review by elisa242003 — APPROVED** [review]
Correct, the changes are well applied, there are no problems

**Pull request merged**